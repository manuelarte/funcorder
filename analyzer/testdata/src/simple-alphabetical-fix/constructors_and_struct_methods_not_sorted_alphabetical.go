package simple_alphabetical_fix

type Greetings struct {
}

func NewOtherGreetings() (m *Greetings) {
	m = &Greetings{}
	return
}

func NewGreetings() *Greetings { // want `constructor \"NewGreetings\" for struct \"Greetings\" should be placed before constructor \"NewOtherGreetings\"`
	return &Greetings{}
}

func (m Greetings) GoodMorning() string {
	return "hello"
}

func (m *Greetings) GoodAfternoon(name string) string { // want `method \"GoodAfternoon\" for struct \"Greetings\" should be placed before method \"GoodMorning\"`
	return "bye"
}

func (m Greetings) hello() string {
	return "hello"
}

func (m *Greetings) bye(name string) string { // want `method \"bye\" for struct \"Greetings\" should be placed before method \"hello\"`
	return "bye"
}
