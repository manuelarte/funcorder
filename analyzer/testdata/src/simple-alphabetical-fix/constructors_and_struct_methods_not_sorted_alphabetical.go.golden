package simple_alphabetical_fix

type Greetings struct {
}

func NewGreetings() *Greetings {
	return &Greetings{}
}

func NewOtherGreetings() (m *Greetings) {
	m = &Greetings{}
	return
}

func (m *Greetings) GoodAfternoon(name string) string {
	return "bye"
}

func (m Greetings) GoodMorning() string {
	return "hello"
}

func (m *Greetings) bye(name string) string {
	return "bye"
}

func (m Greetings) hello() string {
	return "hello"
}
